doc/neps/nep-0054-simd-cpp-highway.rst

```diff
@@ -0,0 +1,480 @@
+
+  - Highway's ``LoadInterleaved2`` would have to be extended with ``*Till``
+    support (only load partial vectors) - that seems doable.
+  - ``Loadn``/``LoadnPair`` can mostly be implemented on top of ``Gather*``,
+    with some specializations.
+
+- ``Lookup128``: the main purpose of this intrinsic is to provide a mapping to
+  the ``_mm512_permutex2var_*`` functions, so it performs 32/bit elements on
+  32-bit data types and 16 elements on 64-bit datatypes; it was need during the
+  replacement of intel SVML with universal intrinsics.
+
+On the other hand, Highway has more instructions that NumPy's universal
+intrinsics, so it's possible that some future needs for NumPy kernels may
+already be met there.
+
+
+Meson changes to be upstreamed (if no Highway runtime dispatching)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Build time detection needs a new module in Meson to detect all the CPU features
+at build time. For the current draft PR implementing that as a new ``feature``
+Meson module, see `meson#11307`_. This is still a decent amount of work -
+probably 2 weeks worth of effort - to complete. It is likely that it will only
+be merged after we prove its robustness inside NumPy. However, it is worth
+pointing out that the initial proposal for SIMD improvements was well-received
+(`meson#11033`), and we expect this to land once it's been proven to work for
+NumPy's needs.
+
+If we'd use Highway including its runtime dispatch features, then we need far
+less from the build system. We only need to know the CPU family, the
+baseline, and the extended features. Meson already has what we need for this
+purpose, e.g. using ``host_machine.cpu_family()`` (see
+`here <https://mesonbuild.com/Reference-tables.html#cpu-families>`__)).
+
+However, given the timelines involved (see the next section) it's not unlikely
+that we'd have to complete the Meson SIMD support for the 1.26.0 release even
+if we choose Highway for 2.0 and beyond.
+
+
+1.26 and 2.0 releases - timing and integration plans
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The plans for our next releases are related, because for Python 3.12 support we
+must use Meson as the build system (because of the removal of ``distutils``),
+and the currently remaining task for completing the migration to Meson is SIMD
+support. The timeline for upcoming releases is:
+
+- Aug 4th, 2023: Python 3.12.0rc1 release date - we need to release NumPy
+  ``1.26.0b1`` (or ``rc1``) around this date, to ensure we don't block
+  downstream projects from using Python 3.12. This pre-release doesn't need
+  SIMD support (performance doesn't matter yet for a beta/rc), although it
+  would be nice to have included already.
+- Oct 4th, 2023: Python 3.12.0 release date - we need a NumPy 1.26.0 release
+  including SIMD support before this date.
+- Dec 31, 2023: NumPy 2.0.0 (could run into January, not a hard deadline)
+
+The current plan is to branch ``maintenance/1.26.x`` off of
```

Comment:
I'm thinking we don't even need a branch, just update the tags and the version in meson.build. As I understand it, the main reason for 1.26 is compatibility tracking.